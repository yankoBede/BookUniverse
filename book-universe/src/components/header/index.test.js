import React from 'react'
import renderer from 'react-test-renderer'
import Header from './index'
import TestingEnvironment from '../../test-utils/router'

describe('Header Component', () => {
    it('should render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value= {{
                user: {
                    loggedIn: true,
                    id: '123'
                }
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })


    it('should render non-authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value= {{
                user: {
                    loggedIn: false
                }
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})