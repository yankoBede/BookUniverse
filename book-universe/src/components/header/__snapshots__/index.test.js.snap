// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render authenticated routes 1`] = `
<nav
  className="nav-bar"
>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/addBook"
      onClick={[Function]}
    >
      Add a new book
    </a>
  </div>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/liked"
      onClick={[Function]}
    >
      My favourite books
    </a>
  </div>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/all"
      onClick={[Function]}
    >
      All books
    </a>
  </div>
  <div
    className="right-list-item"
  >
    <a
      className="link"
      href="/myBooks"
      onClick={[Function]}
    >
      Hello,  undefined
    </a>
  </div>
  <div
    className="right-list-item"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</nav>
`;

exports[`Header Component should render non-authenticated routes 1`] = `
<nav
  className="nav-bar"
>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="left-list-item"
  >
    <a
      className="link"
      href="/top"
      onClick={[Function]}
    >
      Top 5
    </a>
  </div>
  <div
    className="right-list-item"
  >
    <a
      className="link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
  <div
    className="right-list-item"
  >
    <a
      className="link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</nav>
`;
